src/Core/Managed/Shared/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -0,0 +1,170 @@
+﻿namespace Microsoft.ApplicationInsights.DataContracts
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.External;
+    using System.Globalization;
+
+    /// <summary>
+    /// Telemetry type used for availability test results.
+    /// Contains a time and message and optionally some additional metadata.
+    /// </summary>
+    public sealed class AvailabilityTelemetry : ITelemetry, ISupportProperties //TODO: Should this implement IOperation like RequestTelemetry?
```

Comment:
I think it should implement operation telemetry interfaces. The idea may be to associate availability result with the dependency telemetry objects representing every step in multi-step test